index.js

```diff
@@ -55,20 +55,29 @@ module.exports = function(search, replacement, options) {
         callback(null, file);
       }
 
-      if (options && options.skipBinary) {
-        istextorbinary.isText(file.path, file.contents, function(err, result) {
-          if (err) {
-            return callback(err, file);
+      if (options) {
+        if (options.passFilename && typeof replacement === 'function') {
+          var userReplacement = replacement;
+          replacement = function (match) {
+            userReplacement(match, file.path);
```

Comment:
Great idea! But maybe it's better and safer to create `context` object and set its properties explicitly? It will help us with argument immutability.
```javascript
var context = {
  filepath: file.path
}
userReplacement.apply(context, arguments);
```